spec.html

```diff
@@ -28495,12 +28698,19 @@ <h1>
             </ul>
             <p>and it performs FinishLoadingImportedModule(_referrer_, _moduleRequest_, _payload_, _result_) where _result_ is a normal completion, then it must perform FinishLoadingImportedModule(_referrer_, _moduleRequest_, _payload_, _result_) with the same _result_ each time.</p>
           </li>
+          <li>
+            <p>If _moduleRequest_.[[Attributes]] has an entry _entry_ such that _entry_.[[Key]] is *"type"* and _entry_.[[Value]] is *"json"*, the host environment must perform FinishLoadingImportedModule(_referrer_, _moduleRequest_, _payload_, _result_), where _result_ is either the Completion Record returned by an invokation of ParseJSONModule or a throw completion.</p>
```

Comment:
I find writing this as a normative requirement on a host hook roundabout. I see two callsites to HostLoadImportedModule, one in dynamic import and one in static import. WDYT about refactoring those callsites to directly handle JSON modules?

Reply:
@nicolo-ribaudo is this easy to do without significant refactoring? If not, it's fine. Don't have the full machinery in my head.

Reply:
It's a significant refactoring, because right now we don't have any concept of "ask the host to load the source code of a module, but then we parse/initialize it by ourselves". It would need a separate hook.

I reworded a bit the requirement to make it clearer that the host is not expected to simply call that AO: instead it does all its usual stuff, but then _when it calls that AO_ we have some requirements on what parameters it passes to it.

Reply:
Ack, makes sense.